src/ota.rs

```diff
@@ -220,16 +220,43 @@ impl EspOta {
     pub fn initiate_update(&mut self) -> Result<&mut EspOtaUpdate, EspError> {
         self.check_read()?;
 
-        let partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
+        // TODO: The documentation of `esp_ota_get_next_update_partition` states, that the returned
+        // pointer may be a null pointer, which indicates that:
+        // "indicates invalid OTA data partition, or that no eligible OTA app slot partition was found."
+        //
+        // This should probably be handled in here and throw a proper EspError.
+        let update_partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
 
-        let mut handle: esp_ota_handle_t = Default::default();
+        let mut update_handle: esp_ota_handle_t = Default::default();
 
-        esp!(unsafe { esp_ota_begin(partition, OTA_SIZE_UNKNOWN as usize, &mut handle) })?;
+        esp!(unsafe {
```

Comment:
Return back the original code, as we would be dropping the `Option` stuff from above.